src/qibo/result.py

```diff
@@ -437,14 +475,14 @@ def to_dict(self):
         return args
 
     @classmethod
-    def from_dict(cls, payload):
+    def from_dict(cls, payload: dict):
         """Builds a ``CircuitResult`` object starting from a dictionary.
```

Comment:
Missing ``:class:`` reference here and other places. Same thing for some mentions of ``MeasurementOutcomes`` class.